### PR TITLE
refactor(wasm-instrument): add docs, simplify code

### PR DESCRIPTION
**Pull request by StackOverflowExcept1on**
Changes:
```diff
impl SyscallName {
     // removed in favour of `SyscallName::{all, instrumentable}().count()`
-    pub fn count() -> usize { ... }

     // with some code refactoring
-    pub fn instrumentable() -> BTreeSet<Self> { ... }
+    pub fn instrumentable() -> impl Iterator<Item = Self> { ... }

     // not used since wasmi was removed?
-    pub fn to_wgas(self) -> Option<Self> { ... }
}
```

@gear-tech/dev

### CODE REVIEW

File: utils/wasm-instrument/src/rules.rs [hunk 1]
```diff
@@ -1,16 +1,47 @@
+// This file is part of Gear.
```
Comment (NikVolf) [inline utils/wasm-instrument/src/rules.rs:1]: don't include this, it's a fork

Comment (NikVolf) [reply utils/wasm-instrument/src/rules.rs:1]: ah ok, disregard :)

File: utils/wasm-instrument/src/tests.rs [hunk 2]
```diff
@@ -85,85 +83,6 @@ fn prebuilt_simple_module() -> elements::Module {
         .build()
 }
 
-#[test]
```
Comment (NikVolf) [inline utils/wasm-instrument/src/tests.rs:88]: better maybe to replace with test that instrumenting with memory.grow instruction leads to error / panic ?

Comment (StackOverflowExcept1on) [reply utils/wasm-instrument/src/tests.rs:88]: done in f95847462c6b893985d8bcac398b61743862394f

**Review by NikVolf — APPROVED** [review]

**Pull request merged**